File: solutions/two_sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        
        # Iterate through all combinations of numbes with a double for loop that essentially
        # generates a grid of sums based on values from the list, for example:
        # List1 = [1, 3, 5]
        # 
        
        #    1  3  5
        # 1  x  4  6
        # 3  4  x  8
        # 5  6  8  x
        #
        
        for i in range(len(nums)):
            for j in range(len(nums)): # Avoid using the same indexes in iterators to prevent using the same value twice
                if i == j: 
                    break
                test = nums[i] + nums[j] 
                if test == target: # Iterate through all result sums and return the values that add up to the target value
                    return [i, j]

File: golang_solutions/two_sum.go
package main

import "fmt"

// Test main method to feed input and print outupt of my solution function
func main() {

	// Call my function with test values
	var res = twoSum([]int{0, 1, 3, 5, 8, 9}, 10)

	// Display the results
	fmt.Println(res)
}

// Solution to the leetcode problem
// Dummy function for now
func twoSum(nums []int, target int) []int {

	// Nested for loop to iterate through the array twice and compare the numbers if they are a match
	for i := 0; i < len(nums); i++ {
		// Advance the iterator by 1 to avoid comparing a number to itself
		for j := i + 1; j < len(nums); j++ {
			// If a match is found, return an array that contains the indexes of the found numbers
			if nums[i]+nums[j] == target {
				return []int{i, j}
			}
		}
	}
	return nil
}
